src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,42 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Deadline class to handle tasks that are need to be done before a specific date/time
+ */
+public class Deadline extends Task {
+    protected final LocalDate by;
+
+    /**
+     * Constructor for DeadLine class
+     *
+     * @param description Details of the task
+     * @param by Specific date/time to complete the task
+     */
+    public Deadline(String description, LocalDate by) {
+        super(description);
+        this.by = by;
+    }
+
+    /**
+     * Displays simplified version of task type, description and deadline of tasks in Duke.txt
+     *
+     * @return String format regarding the task deadline information
+     */
+    @Override
+    public String formatTask() {
+        return String.format("D | %s", super.formatTask());
+    }
+
+    /**
+     * Displays task type, description and deadline of task
+     *
+     * @return String format regarding the task deadline information
+     */
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + this.by.format(DateTimeFormatter.ofPattern("dd-MM-yyyy")) + ")";
```

Comment:
Hi, breaking up this line could make your program more readable. According to the coding standards, it will be best to have line length <=120 (Hard Limit) and <110 (Soft Limit). The character for this line is around 118 so it might be better to break this line up instead.